src/core/thread/mle.cpp

```diff
@@ -1061,7 +1061,9 @@ otError Mle::AppendLeaderData(Message &aMessage)
 
 void Mle::FillNetworkDataTlv(NetworkDataTlv &aTlv, bool aStableOnly)
 {
-    uint8_t length;
+    uint8_t length = 255; // sizeof( NetworkDataTlv::mNetworkData )
```

Comment:
Well, since this `length` initial value should actually by the size of the buffer we pass down, it should be `sizeof( NetworkDataTlv::mNetworkData )`, which is a private member, the length should be exactly as defined there. Keeping `kMaxSize` in one place but not the other will be more misleading, I'm afraid.

What about changing `https://github.com/openthread/openthread/blob/master/src/core/thread/mle_tlvs.hpp#L844` to